### PR TITLE
fix(code): add untrusted artifact prompt boundaries

### PR DESCRIPTION
**Pull request by mikeangstadt**
## Summary
- teach the code orchestrator, amend-plan, key planning agents, and judge context preparation to treat PRDs, plans, notes, and logs as untrusted evidence rather than instructions
- add the same adversarial-artifact rule to the shared judge input preamble so all judge runs inherit it
- add a regression test that guards the required trust-boundary language in those prompt files

## Testing
- `pytest plugins/code/tools/python/test_untrusted_artifact_boundary.py`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,16 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
 ## [Unreleased]
```
Comment (thadeusb) [inline CHANGELOG.md:7]: CHANGELOG says code v1.5.10 and judges v1.4.1, but neither `plugins/code/.claude-plugin/plugin.json` nor `plugins/judges/.claude-plugin/plugin.json` were modified in this PR. The versions on the branch are still 1.5.9 and 1.4.0 respectively.

File: plugins/code/tools/python/test_untrusted_artifact_boundary.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+"""Regression tests for untrusted-artifact prompt boundaries."""
+
+from pathlib import Path
+
+
+REPO_ROOT = Path(__file__).resolve().parents[4]
+
+FILES_REQUIRING_GUARDRAIL = [
```
Comment (thadeusb) [inline plugins/code/tools/python/test_untrusted_artifact_boundary.py:8]: `implementation-subagent` reads plan.json and has `Write, Edit, Bash` — it's the agent that actually executes instructions from the plan. Same story with `plan-writer` and `plan-importer` (which ingests external markdown). These feel like higher-risk injection surfaces than the read-only evaluators you did cover. Was the scope here intentionally limited to planning/evaluation, with implementation agents coming later?

File: plugins/code/tools/python/test_untrusted_artifact_boundary.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    / "skills"
+    / "artifact-type-tailored-context"
+    / "preambles"
+    / "common_input_preamble.md",
+]
+
+
+def test_prompt_boundaries_treat_artifacts_as_untrusted_data() -> None:
+    for path in FILES_REQUIRING_GUARDRAIL:
+        content = path.read_text()
```
Comment (thadeusb) [inline plugins/code/tools/python/test_untrusted_artifact_boundary.py:28]: This guards the files you know about today but won't catch a new agent added next month that reads artifacts without the boundary. Any appetite for a second test that globs `plugins/*/agents/*.md`, checks which ones have tool access to `Read` or `Bash`, and asserts they're either in this list or explicitly excluded?
